Library/ViewModels/PledgeViewModel.swift

```diff
@@ -449,87 +446,115 @@ public class PledgeViewModel: PledgeViewModelType, PledgeViewModelInputs, Pledge
-        .filter { $0.isTerminating }
-        .ignoreValues(),
-      createPaymentAuthorizationDidFinishSignal
-    )
+    let scaFlowCompletedWithError = self.scaFlowCompletedWithResultSignal
+      .filter { $0.0.status == .failed }
+      .map(second)
+      .skipNil()
 
-    let createApplePayBackingEventSuccess = createApplePayBackingEvent.values()
+    let scaFlowCompletedWithSuccess = self.scaFlowCompletedWithResultSignal
+      .filter { $0.0.status == .succeeded }
+      .map(first)
+      .ignoreValues()
 
-    let createApplePayBackingEventErrors = createApplePayBackingEvent.errors()
-      .map { $0.localizedDescription }
+    let didInitiateApplePayBacking = Signal.merge(
+      willCreateApplePayBacking,
+      willUpdateApplePayBacking
+    )
 
-    let createApplePayBackingError = createApplePayBackingEventErrors
-      .takeWhen(createApplePayBackingCompleted)
+    let paymentAuthorizationDidFinish = didInitiateApplePayBacking
+      .takeWhen(self.paymentAuthorizationDidFinishSignal)
 
-    self.createBackingError = Signal.merge(
-      createApplePayBackingError,
-      createBackingEventError
+    let createOrUpdateApplePayBackingCompleted = Signal.combineLatest(
+      Signal.merge(
+        updateBackingEvent.filter { $0.isTerminating }.ignoreValues(),
+        createBackingEvent.filter { $0.isTerminating }.ignoreValues()
+      ),
+      paymentAuthorizationDidFinish
     )
 
-    let applePayTransactionCompleted = Signal.combineLatest(project, createApplePayBackingEventSuccess)
-      .takeWhen(createApplePayBackingCompleted)
-      .map(first)
+    let createOrUpdateBackingEventValues = Signal.merge(
+      createBackingEvent.values().map { $0 as StripeSCARequiring },
+      updateBackingEvent.values().map { $0 as StripeSCARequiring }
+    )
 
-    let createBackingTransactionSuccess = project.takeWhen(createBackingEventSuccess)
+    let createOrUpdateBackingEventValuesNoSCA = createOrUpdateBackingEventValues
+      .filter(requiresSCA >>> isFalse)
 
-    self.goToThanks = Signal.merge(applePayTransactionCompleted, createBackingTransactionSuccess)
+    let createOrUpdateBackingDidCompleteNoSCA = submitButtonTapped
+      .takeWhen(createOrUpdateBackingEventValuesNoSCA)
+      .filter(isTrue)
+      .ignoreValues()
 
-    // MARK: - Success/Failure Update
+    let createOrUpdateBackingEventValuesRequiresSCA = createOrUpdateBackingEventValues
+      .filter(requiresSCA)
 
-    let updatePaymentAuthorizationDidFinishSignal = willUpdateApplePayBacking
-      .takeWhen(self.paymentAuthorizationDidFinishSignal)
+    self.beginSCAFlowWithClientSecret = createOrUpdateBackingEventValuesRequiresSCA.map { $0.clientSecret }
+      .skipNil()
 
-    let updateApplePayBackingCompleted = Signal.combineLatest(
-      updateBackingEvent
-        .filter { $0.isTerminating }
-        .ignoreValues(),
-      updatePaymentAuthorizationDidFinishSignal
-    )
+    let didCompleteApplePayBacking = createOrUpdateBackingEventValues
```

Comment:
Curious why do we need to take `createOrUpdateBackingEventValues` here? That seems to require either `createBackingEvent.values()` or `updateBackingEvent.values()`.

Reply:
Yes this is future proof for when `createBacking` supports Apple Pay.